File: VIMNetworking/src/main/java/com/vimeo/networking/model/Upload/UploadQuota.java
package com.vimeo.networking.model.Upload;


import com.vimeo.networking.Vimeo;

/**
 * Created by kylevenn on 8/19/15.
 */
public class UploadQuota {

    public Space space;
    public Quota quota;

    // Returns -1 if there is no space object on this user
    public long getFreeUploadSpace() {
        if (space != null) {
            return space.free;
        } else {
            return Vimeo.NOT_FOUND;
        }
    }
}

File: models/src/main/java/com/vimeo/networking2/UploadQuota.kt
package com.vimeo.networking2

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * User's upload quota information.
 *
 * @param lifetime The number of bytes remaining in your lifetime maximum.
 * @param periodic The number of bytes remaining in your upload quota for the current period.
 * @param quota Quota information.
 * @param space Space information.
 */
@JsonClass(generateAdapter = true)
data class UploadQuota(

    @Json(name = "lifetime")
    val lifetime: Lifetime? = null,

    @Json(name = "periodic")
    val periodic: Periodic? = null,

    @Json(name = "quota")
    val quota: Quota? = null,

    @Json(name = "space")
    val space: Space? = null

)
